backend/django/create-model-and-api.mo.md

```diff
@@ -0,0 +1,219 @@
+```
+
+- Finally, make your migrations and migrate.
+
+
+### Add the News model to the admin *(~2 min)*
+
+- We now need to add the model to the admin back-office.
+
+ourproject/publications/admin.py
+```python
+from django.contrib import admin
+from .models import News
+
+
+@admin.register(News)
+class MyNewsAdmin(admin.ModelAdmin):
+    list_display = (
+        'title',
+        'city'
+    )
+```
+
+If you don't override `list_display` by writing `pass` instead of `list_display`, the default column title in the admin will be the __str__ returned in the model.
+list_display allows you to add several columns to the admin interface.
+
+
+### Serialize the News you get from the database *(~2 min)*
+
+- Serializers define the API representation.
+
+
+ourproject/publications/serializers.py
+```python
+from .models import News
+from rest_framework import serializers
+
+
+class NewsSerializer(serializers.HyperlinkedModelSerializer):
+    id = serializers.IntegerField(source='pk')
+    city = serializers.StringRelatedField(many=False, source="city.pk")   #We need this to get the primary key of the city that is attached to this news
+
+    class Meta:
+        model = News
+        fields = ('title', 'description', 'created_at', 'updated_at', 'city', 'id')    #All the fields you wish to get
+```
+
+
+### Create a News ViewSet *(~2 min)*
+
+
+```python
+from .models import News
+from rest_framework import viewsets
+from .serializers import NewsSerializer
+
+#This is an example of filtering the pieces of news with the city-id like this: /?city-id=5
+class FilterByCity(object):   
+    def get_queryset(self):   #You need to override the default get_queryset method
+        queryset = super().get_queryset()
+        city_id = self.request.query_params.get('city-id')
+        if city_id is not None:
+            queryset = queryset.filter(city=city_id)
+        return queryset
+
+
+class NewsViewSet(FilterByCity, viewsets. ):
+    queryset = News.objects.all()
+    serializer_class = NewsSerializer
+```
+
+- The NewsViewSet class will now inherit both classes FilterByCity viewsets.ModelViewSet from right to left. The FilterByCity get_queryset method will override the ModelViewSet one because the class is the furthest at the left.
+
+
+
+### Create a News ViewSet *(~2 min)*
+
+- Create a route for the ViewSet.
+
+ouproject/config/router.py
+```python
+from rest_framework import routers
+from intramuros.users.viewsets import UserViewSet
+from intramuros.locations.viewsets import CityViewSet
+from intramuros.publications.viewsets import NewsViewSet  #Add this line
+
+
+# Routers provide an easy way of automatically determining the URL conf.
+router = routers.DefaultRouter()
+router.register(r'users', UserViewSet)
+router.register(r'locations/cities', CityViewSet)
+router.register(r'publications/news', NewsViewSet)    #Add this one too
+```
+
+- You are good to go! :D
+
+- If you go on this url: HOST/publications/news/
```

Comment:
Put a new title above for : # Result ? :)

Reply:
done :)